peachjam/views/courts.py

```diff
@@ -30,20 +35,42 @@ def get_queryset(self):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
-        years = self.model.objects.filter(court=self.court).dates(
-            "date", "year", order="DESC"
-        )
+        context["doc_type"] = "Judgment"
+        context["court"] = self.court
+        context["formatted_court_name"] = self.formatted_court_name()
+        context["registries"] = self.court.registries.exclude(
+            judgments__isnull=True
+        )  # display registries with judgments only
+        context["registry_groups"] = list(chunks(context["registries"], 2))
+        context["court_detail_view"] = True
 
-        context["years"] = [
-            {
-                "url": reverse("court_year", args=[self.court.code, y.year]),
-                "year": y.year,
-            }
-            for y in years
-        ]
+        if "year" in self.kwargs:
+            context["court_detail_view"] = False
 
-        context["all_years_url"] = reverse("court", args=[self.court.code])
+            context["year"] = self.kwargs["year"]
+
+            if "month" in self.kwargs:
+                context["month"] = MONTHS[self.kwargs["month"]]
+                context[
+                    "formatted_court_name"
+                ] = f"{context['formatted_court_name']} - {context['month']} {self.kwargs['year']}"
+            else:
+                context[
+                    "formatted_court_name"
+                ] = f"{context['formatted_court_name']} - {self.kwargs['year']}"
+
+            context["all_months_url"] = reverse(
+                "court_year", args=[self.court.code, self.kwargs["year"]]
+            )
+            context["documents"] = self.grouped_judgments(self.get_base_queryset())
```

Comment:
this should always happen, not just for a year or month. Also, the grouping should happen on the main document list that is already in the context, thanks to the super class.